circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/NavigableCircuitContent.kt

```diff
@@ -50,11 +55,30 @@ public fun NavigableCircuitContent(
     )
 
   if (backstack.size > 0) {
-    @Suppress("SpreadOperator")
     decoration.DecoratedContent(activeContentProviders, backstack.size, modifier) { provider ->
-      val values = providedValues[provider.record]?.provideValues()
-      val providedLocals = remember(values) { values?.toTypedArray() ?: emptyArray() }
-      CompositionLocalProvider(*providedLocals) { provider.content(provider.record) }
+      // We retain the record's retained state registry if the back stack
+      // contains the record
+      val record = provider.record
+      val retainChecker =
+        remember(backstack, record) { CanRetainChecker { backstack.contains(record) } }
+
+      CompositionLocalProvider(LocalCanRetainChecker provides retainChecker) {
```

Comment:
This retainChecker is only used for the retained registry. I don't love using a `CompositionLocalProvider` to provide a value to a single caller, but the alternative was adding a `CanRetainChecker` param to `rememberRetained`, which didn't feel like it should be public API.